### DISCUSSION TITLE
Register user without email

### DISCUSSION THREAD

**Discussion by pjsde** [codeigniter4/shield#193] [Q&A]
Hello,

I know that there is an assumption that the user has an email to be able to register, but if the specifications of a particular project say that the user does not have an email and that his identification is based on the username/password, how can I implement this situation?

**Comment by kenjis** [comment]
It is the opposite of #192, but `email` is required by this package.
So it seems you need to customize the source code that does not work.

**Comment by lonnieezell** [comment]
How would this work? How does this system verify users? Is it phone based primarily? Just trying to understand the use case.

**Comment by pjsde** [comment]
@lonnieezell Yes, the verification it will be based on sms messages

**Comment by EllyeFS** [comment]
Is there still no built-in way for users to register without an e-mail?

**Reply by kenjis** [reply]
For now, there are no plans to offer registration without email, as the system is designed to have email.

How will the system work without email?

**Reply by EllyeFS** [reply]
The e-mail verification feature is already optional anyway - it wouldn't be so different then just how it works with it turned off.

And yeah, this would mean no account recovery. My idea would be to make it optional like some sites do (Reddit used to, for example): you don't need to input an email, but if you don't you can't recover your password.

But not a big deal for me if it can't be like that, no worries.
